src/connectedk8s/azext_connectedk8s/_validators.py

```diff
@@ -17,3 +21,10 @@ def example_name_or_id_validator(cmd, namespace):
                 type='storageAccounts',
                 name=namespace.storage_account
             )
+
+def validate_private_link_properties(namespace):
+    if (not namespace.enable_private_link or namespace.enable_private_link == "false") and namespace.private_link_scope_resource_id:
+        raise CLIError("Conflicting private link parameters received. The parameter '--private-link-scope-resource-id' should not be set if '--enable-private-link' is passed as null or False.")
+    if namespace.enable_private_link == "true" and not namespace.private_link_scope_resource_id:
+        raise CLIError("The parameter '--private-link-scope-resource-id' was not provided. It is mandatory to pass this parameter for enabling private link on the connected cluster resource.")
```

Comment:
ArgumentUsageError instead on CLIError

`from azure.cli.core.azclierror import ArgumentUsageError`

Reply:
changed to ArgumentUsageError.